### DISCUSSION TITLE
Blurred result when stitching one image after another

### DISCUSSION THREAD

**Discussion by cxy-ccc** [OpenStitching/stitching#84] [Q&A]
When i try to stitch 20 images, an error occurred. The code only stitched 16 images.
Traceback (most recent call last):
  File "/home/zx/tiegui/stitching-main/setup.py", line 53, in <module>
    panorama2 = stitcher2.stitch([imgA,imgB])
  File "/home/zx/tiegui/stitching-main/stitching/stitcher.py", line 92, in stitch
    imgs, features, matches = self.subset(imgs, features, matches)
  File "/home/zx/tiegui/stitching-main/stitching/stitcher.py", line 138, in subset
    matches,
  File "/home/zx/tiegui/stitching-main/stitching/subsetter.py", line 28, in subset
    indices = self.get_indices_to_keep(features, matches)
  File "/home/zx/tiegui/stitching-main/stitching/subsetter.py", line 53, in get_indices_to_keep
    raise StitchingError("No match exceeds the " "given confidence threshold.")
stitching.stitching_error.StitchingError: No match exceeds the given confidence threshold.
But the image has enough overlapping region.
The result of 16 images and the first half is vague.
![20230202113005607](https://user-images.githubusercontent.com/53075820/218052221-4372807a-dd82-40ab-a2dd-7f8165c1cc19.png)
When stitch this image, the error occurred.
![20230202113007191](https://user-images.githubusercontent.com/53075820/218052849-d910f920-24d8-48e0-bdaa-0f152598934a.png)

**Comment by lukasalexanderweber** [comment]
Can you share the images?

**Comment by lukasalexanderweber** [comment]
`panorama2 = stitcher2.stitch([imgA,imgB])` dont you stitch just two Images?

**Comment by cxy-ccc** [comment]
- [ ] 

> 

I stitch one piece after another. The images are arranged in order.
I have no idea why the first half is vague.

**Comment by lukasalexanderweber** [comment]
Can't you just collect all Images in a List and than Stitch?

**Comment by cxy-ccc** [comment]
> Can't you just collect all Images in a List and than Stitch?

This may cost more time. Now the error was solved, but i don't know why the first half is so vague.

**Reply by Greatuso** [reply]
Hi,I met the same situation,have you found the cause of this problem？

**Comment by lukasalexanderweber** [comment]
Try Just for the image quality to Stitch all Images together, so we can see If its vague as well

**Comment by cxy-ccc** [comment]
> Try Just for the image quality to Stitch all Images together, so we can see If its vague as well

The quality is high, 1280*1280. And I didn't compress the size of image.

**Comment by lukasalexanderweber** [comment]
Can you share the images?

**Comment by lukasalexanderweber** [comment]
You can reduce the image size for testing with the final_megapix parameter. So you don't have to wait to long for a result

**Comment by KeineAhnung1337** [comment]
Hi, I have the same issue.
If I put all into a list, it takes ages to calculate and gets stuck at 4th step (03_matches_graph is made).
If i do it in intervalls of two recursive (e.g. 80 pictures into 40 into 20 into 10 into 5 into 2+1). If i put the list of 5 into the function, it crashes as well caused by a camera issue.
It is very blurry since I let it stitch for 5 times.
Is there any fix/parameter I can change? What are the detector options? I did not find anything in the docu.

**Reply by lukasalexanderweber** [reply]
Can you share sample Images?
